src/sqlfluff/core/rules/base.py

```diff
@@ -145,10 +149,10 @@ def __init__(self, edit_type, anchor: BaseSegment, edit=None):
                         "Developer Note: Edit segment found with preset position marker. "
                         "These should be unset and calculated later."
                     )
-                    seg.pos_marker = None
-        # Once stripped, we shouldn't replace any markers because
-        # later code may rely on them being accurate, which we
-        # can't guarantee with edits.
+                    seg.pos_marker = None  # type: ignore
```

Comment:
existing type error, previously this wasn't getting checked as this class wasn't type hinted.
Had to type ignore this line as this is being typed elsewhere as `PositionMarker`: https://github.com/sqlfluff/sqlfluff/blob/7fd452dbfb9ab8537b8ca6e92c9489ff04d6172b/src/sqlfluff/core/parser/segments/base.py#L123

I tried changing that to `Optional[PositionMarker]` but that raised even more type errors as other areas of the code base aren't taking into account that seg.pos_marker could be set to None here for BaseSegments.

Basically this PR changes nothing here, so we can leave as is and investigate afterwards 👍